File: scala/grains/src/main/scala/Grains.scala
object Grains {
  private val squares = 64

  def square(n: Int): Option[BigInt] = {
    if (n < 1 || n > squares)
      None
    else
      Some(BigInt.int2bigInt(1) << (n - 1))
  }

  def total: BigInt = (1 to squares).map(square(_).get).sum
}

File: ruby/grains/grains.rb
class Grains
  SQUARE_COUNT = 64

  def self.square(square)
    raise ArgumentError, 'Invalid square' if
      square < 1 || square > SQUARE_COUNT

    1 << (square - 1)
  end

  def self.total
    (1..SQUARE_COUNT).sum { |i| Grains.square(i) }
  end
end
